### ISSUE TITLE
How do I ban the camera editor from players?

### ISSUE THREAD

**Issue by MugGod** [mchorse/aperture#63]
The mod can be used on the server - it's cool. But for some reason, there are still no permissions in fashion to open the camera editor. At a minimum, it is necessary that the "/op" requirement should be enabled to open it. But since there is no such requirement, any players can open the editor on the server and use it as an Xray cheat.
Is there a way to restrict access to the camera editor?
p.s. Your mod is awesome. User from Russia

**Comment by mchorse** [comment]
It should be already a feature. Check Aperture `1.6` or above. It shouldn't let users go into camera editor on a server. However, if you still can open the camera editor in non-OP, then go to Ctrl + 0 (my mod options), press ⬆ looking icon, OP access, and disable Camera Editor option.

Hopefully that helps!

P.S.: привет 👋

**Comment by MugGod** [comment]
Thank you for such a quick response! I did as you said. First I executed the OP command to access the "OP Access" settings, then I disabled the Camera Editor function in the settings, after that I executed the DEOP command to check if it works, unfortunately the interface still keeps opening. I also tried restarting the client and server, I opened the settings again and checked if they were reset, all settings are saved, but for some reason players without OP can still open the camera editor. I also noticed that sometimes when opening the interface in the chat, the message "OpenGL error: 1282 (invalid operation)" may appear, despite this error, the interface still opens and works completely.
I want to draw attention to the fact that the server runs on Sponge core, it is he who loads the mods, I do not know if this is related. 
My Forge: forge-1.12.2-14.23.5.2838
![image](https://user-images.githubusercontent.com/82241098/138165362-e945200b-32db-44fa-940d-79f3ff45aad0.png)
![image](https://user-images.githubusercontent.com/82241098/138165391-ef833e9b-1d6d-4e6b-bc96-f6d89fece6f8.png)

**Comment by MugGod** [comment]
Well, after looking into the situation for a long time, I found out that the problem is that the Sponge has OP functions disabled, it only works with permissions, this is clearly a mod problem, the mod needs to add a special permission to open the camera editor interface. Is it difficult?

**Comment by mchorse** [comment]
Ah… my bad, I forgot to respond back then. 😥 

Even if it’s a mod problem, I won’t try fixing it. I quit Minecraft modding, and just replying to any questions about mods whenever possible. Sorry for this being unhelpful.

**Comment by MugGod** [comment]
This is very sad, because there are no normal analogues of your mod. The only CMDcam mod that provides similar functionality does not support even simple cuts in cutscenes. When I wrote to the developer of this mod with this question about slicing, they advised me this mod, saying that it fits better.

**Comment by MugGod** [comment]
ಥ_ಥ

**Comment by Chryfi** [comment]
Wait, so is it Sponge's fault? Because Sponge does weird things with OP? Did I understand that correctly?

**Comment by MugGod** [comment]
> 

Sponge doesn't use the OP system at all, it uses Permissions because the permission system is always the basis of servers, OP is rarely used on servers. Therefore, the OP on the Sponge is disabled. therefore, it is usually the permissions system that is used in mods and plugins. (つ⩹ᴥ⩺)つ

**Comment by Chryfi** [comment]
Then I have to disappoint you. This is clearly NOT the mods problem. It is Sponge's fault and normal forge server don't have this problem.

**Comment by MugGod** [comment]
> 

permissions are not used on regular forge servers, because there are no plugins in forge that allow you to manage them, why do you think mods use permissions and why do mod developers add them at all? That's what they're used for, because the servers don't use an OP. It doesn't support groups and fine-tuning, it's just an outdated thing. I understand that many mod developers do not approve of using their mods on servers, this is not the first time I have been written like this, it's sad. Yes, this is not a problem of mods, but it is not a problem of servers either, because it is simply impossible to manage the server using a primitive OP, and no one is doing this. Server cores can't do anything about the fact that OP is primitive and doesn't support what permissions supports. Therefore, the only hope is that mod developers will adapt mods for permissions. Only modders can do something. There is only hope for you. No one forces you to do this, of course. :(

Sponge just disabled the OP function because it is considered useless for servers because permissions are commonly used.
anyway, thanks for answering.

**Comment by mchorse** [comment]
I would agree with MudGod that permission system is much better than simply OP vs non-OP users (because you can have moderators, developers, and other kind of stuff that need only specific access to plugin functionality).

But IMO it sort of Sponge's fault for "breaking" OP functionality (even though it was done in a good faith).

> Therefore, the only hope is that mod developers will adapt mods for permissions.

~~There is no easy way to integrate permission API into Forge mods. Honestly, I don't even sure if it's possible.~~ I found [here](https://github.com/SpongePowered/SpongeDocs/pull/621/commits/6879b2d0736a8cdc7a9b3674a87813dd1bacc485) an interesting way to being able to check for Sponge permissions through vanilla code, but there is no guarantee whether it's going to work at all. As I said earlier, I'm no longer work on my mods. If someone is willing look into it, feel free to.

**Comment by MugGod** [comment]
> 

Yeap, I understand you, thank you for the answer! Your mod is still awesome ✨
